File: typescript/lib/openrag-utils/documents.ts
/**
 * OpenRAG Documents Utilities
 * 
 * Functions for document ingestion and deletion.
 * Can be run as a standalone script to test document endpoints.
 */

import { OpenRAGClient, IngestResponse, DeleteDocumentResponse } from 'openrag-sdk';

/**
 * Ingest a document into the knowledge base.
 * 
 * @param filePath - Path to the file to ingest (Node.js)
 * @param file - File or Blob object to ingest (Browser)
 * @param filename - Filename for the document
 * @param wait - Whether to wait for ingestion to complete (default: true)
 * @param client - Optional OpenRAGClient instance
 * @returns Ingestion result with status and task_id
 */
export async function ingestDocument(
  options: {
    filePath?: string;
    file?: File | Blob;
    filename?: string;
    wait?: boolean;
  },
  client?: OpenRAGClient
): Promise<IngestResponse> {
  const useClient = client || new OpenRAGClient();
  
  return await useClient.documents.ingest({
    filePath: options.filePath,
    file: options.file,
    filename: options.filename,
    wait: options.wait ?? true
  });
}

/**
 * Delete a document from the knowledge base.
 * 
 * @param filename - The filename to delete
 * @param client - Optional OpenRAGClient instance
 * @returns Deletion result with success status
 */
export async function deleteDocument(
  filename: string,
  client?: OpenRAGClient
): Promise<DeleteDocumentResponse> {
  const useClient = client || new OpenRAGClient();
  return await useClient.documents.delete(filename);
}

/**
 * Test document endpoints (for standalone execution)
 */
async function main() {
  const apiKey = process.env.OPENRAG_API_KEY;
  const url = process.env.OPENRAG_URL;

  console.log(`Using API Key: ${apiKey?.slice(0, 20) || 'NOT SET'}...`);
  console.log(`Using URL: ${url}\n`);

  // Note: This is a placeholder test - you would need an actual file to test
  console.log('=== Document Operations ===');
  console.log('To test document ingestion:');
  console.log('  const result = await ingestDocument({ filePath: "./test.pdf" });');
  console.log('  console.log(`Status: ${result.status}`);');
  console.log();
  console.log('To test document deletion:');
  console.log('  const result = await deleteDocument("test.pdf");');
  console.log('  console.log(`Success: ${result.success}`);');
  console.log();
}

// Allow running as standalone script
if (require.main === module) {
  main().catch(console.error);
}

// Made with Bob

File: python/openrag_utils/documents.py
"""
OpenRAG Documents Utilities

Functions for document ingestion and management.
Can be run as a standalone script to test document endpoints.
"""
import asyncio
from pathlib import Path
from typing import Optional

from openrag_sdk import OpenRAGClient

from config import config


async def ingest_document(
    file_path: str,
    wait: bool = True,
    client: Optional[OpenRAGClient] = None
) -> dict:
    """
    Ingest a document into the knowledge base.

    Args:
        file_path: Path to the file to ingest
        wait: Whether to wait for ingestion to complete (default: True)
        client: Optional OpenRAGClient instance

    Returns:
        dict: Ingestion result with status, task_id, successful_files, failed_files
    """
    if client:
        result = await client.documents.ingest(file_path=file_path, wait=wait)
        return {
            "status": result.status,
            "task_id": result.task_id,
            "successful_files": result.successful_files,  # type: ignore
            "failed_files": result.failed_files  # type: ignore
        }

    async with OpenRAGClient() as client:
        result = await client.documents.ingest(file_path=file_path, wait=wait)
        return {
            "status": result.status,
            "task_id": result.task_id,
            "successful_files": result.successful_files,  # type: ignore
            "failed_files": result.failed_files  # type: ignore
        }


async def delete_document(
    filename: str,
    client: Optional[OpenRAGClient] = None
) -> bool:
    """
    Delete a document from the knowledge base.

    Args:
        filename: Name of the file to delete
        client: Optional OpenRAGClient instance

    Returns:
        bool: True if successful
    """
    if client:
        result = await client.documents.delete(filename)
        return result.success

    async with OpenRAGClient() as client:
        result = await client.documents.delete(filename)
        return result.success


async def main():
    """Test document endpoints"""
    api_key = config.OPENRAG_API_KEY
    url = config.OPENRAG_URL

    print(f'Using API Key: {api_key[:20] if api_key else "NOT SET"}...')
    print(f'Using URL: {url}\n')

    # Create a test file
    test_file = Path("test_document.md")
    test_file.write_text("# Test Document\n\nThis is a test document for OpenRAG ingestion.")

    try:
        # Ingest document
        print('=== Ingest Document ===')
        print(f"Ingesting: {test_file}")
        result = await ingest_document(str(test_file), wait=True)
        print(f"Status: {result['status']}")
        print(f"Task ID: {result['task_id']}")
        print(f"Successful: {result['successful_files']}")
        print(f"Failed: {result['failed_files']}")
        print()

        # Delete document
        print('=== Delete Document ===')
        print(f"Deleting: {test_file.name}")
        success = await delete_document(test_file.name)
        print(f"Success: {success}")
        print()

    except Exception as e:
        print(f"Error: {e}")
    finally:
        # Clean up test file
        if test_file.exists():
            test_file.unlink()
            print(f"Cleaned up test file: {test_file}")


if __name__ == '__main__':
    asyncio.run(main())

# Made with Bob
